packages/provider/src/api/interfaces/endpoint/endpoint.ts

```
@@ -0,0 +1,25 @@
+// Copyright 2021-2024 Prosopo (UK) Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+import type { ApiResponse } from "@prosopo/types";
+import type { ZodType, z } from "zod";
+
+interface Endpoint<T extends ZodType | undefined> {
```

Is it possible for us to not have undefined here? We should be able to throw earlier if undefined or something?

---

It was an intentional thing. 

In my view, potentially we can have endpoints without input arguments. For example: giveCurrentServerState

The current declaration allows the use of this interface in that case, allowing the endpoint to decide, whether to request arguments or not